### PR TITLE
Introduce Site Verification module and add support for internal modules

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Addresses issue #407

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/modules-list.js [hunk 1]
```diff
@@ -73,11 +72,17 @@ class ModulesList extends Component {
...
+		// Map of slug => name for every module that is active and completely set up.
+		const completedModuleNames = {};
 
 		// Sort Modules by sort order.
-		let sortedModules = sortBy( modules, ( module, key ) => {
-			module.key = key;
+		let sortedModules = sortBy( modules, ( module ) => {
+			if ( module.active && module.setupComplete ) {
+				completedModuleNames[ module.slug ] = module.name;
+			}
```
Comment (aaemnnosttv) [inline assets/js/components/modules-list.js]: `completedModuleNames` shouldn't be mutated within a callback for sorting logic.

This operation should probably be done as part if the initialization of `completedModuleNames`,
e.g
```js
const completedModuleNames = modules
	.filter( ( module ) => module.active && module.setupComplete )
	.reduce( ( completed, module ) => {
		completed[ module.slug ] = module.name;
		return completed;
	}, {} );
```

Then `sortedModules` could be simplified as well 

```js
const sortedModules = _( modules )
	.filter( ( module ) => 0 === module.required.length )
	.sortBy( 'sort' )
	.value();
```

File: assets/js/components/modules-list.js [hunk 2]
```diff
@@ -73,11 +72,17 @@ class ModulesList extends Component {
 	}
 
 	render() {
-		const modules = window.googlesitekit.modules || [];
+		// Filter out internal modules.
+		const modules = filter( window.googlesitekit.modules || [], ( module ) => ! module.internal );
```
Comment (aaemnnosttv) [inline assets/js/components/modules-list.js]: Consider using [`reject`](https://lodash.com/docs/4.17.14#reject) instead.
I.e. `reject( modules, ( module ) => module.internal )`

File: assets/js/components/modules-list.js (hunk 1 above)
Comment (felixarntz) [reply assets/js/components/modules-list.js]: Makes sense, I'll update that. We can probably use native array functions `filter()`, `sort()` though right?

File: assets/js/components/modules-list.js (hunk 2 above)
Comment (felixarntz) [reply assets/js/components/modules-list.js]: I'd prefer to use a native array function instead, if we make these changes. `filter()` would work similarly to Lodash's one.

File: assets/js/components/modules-list.js (hunk 1 above)
Comment (aaemnnosttv) [reply assets/js/components/modules-list.js]: Yeah, the first block is all native functions, the second could be rewritten to use native as well, we'd just need to implement the sorting.

**Comment by felixarntz** [comment]
> Everything looks good, I just found one instance left over in the `Analytics` module which needs to be simply removed I think.
> 
> https://github.com/google/site-kit-wp/blob/1956c5998fa15561b211561cb1466f016fdc5827/includes/Modules/Analytics.php#L39-L47
> 
> Other than that, it looks like everything has been relocated to the new module 👍

@aaemnnosttv This is actually unrelated (it's in the Analytics module). It is an older "copy-paste" missing piece. #83 already will remove it, so I'd prefer to let it be.

**Pull request merged**